cygwin/CygwinProcessTable.c

```diff
@@ -0,0 +1,799 @@
+            char tmpCommandChar = command[i];
+
+            if (command[i] <= ' ') {
+               if (!tokenEnd) {
+                  command[i] = '\0';
+
+                  bool found = Compat_faccessat(AT_FDCWD, command, F_OK, AT_SYMLINK_NOFOLLOW) == 0;
+
+                  // Restore if this wasn't it
+                  command[i] = found ? '\n' : tmpCommandChar;
+
+                  if (found)
+                     tokenEnd = i;
+                  if (!tokenArg0Start)
+                     tokenArg0Start = tokenStart;
+               } else {
+                  // Split on every further separator, regardless of path correctness
+                  command[i] = '\n';
+               }
+            } else if (!tokenEnd) {
+               if (command[i] == '/' || (command[i] == '\\' && (!tokenStart || command[tokenStart - 1] == '\\'))) {
+                  tokenStart = i + 1;
+               } else if (command[i] == ':' && (command[i + 1] != '/' && command[i + 1] != '\\')) {
+                  tokenEnd = i;
+               }
+            }
+         }
+
+         if (!tokenEnd) {
+            tokenStart = tokenArg0Start;
+
+            // No token delimiter found, forcibly split
+            for (int i = 0; i <= lastChar; i++) {
+               if (command[i] <= ' ') {
+                  command[i] = '\n';
+                  if (!tokenEnd) {
+                     tokenEnd = i;
+                  }
+               }
+            }
+         }
+      }
+
+      /* Some command lines are hard to parse, like
+       *   file.so [kdeinit5] file local:/run/user/1000/klauncherdqbouY.1.slave-socket local:/run/user/1000/kded5TwsDAx.1.slave-socket
+       * Reset if start is behind end.
+       */
+      if (tokenStart >= tokenEnd)
+         tokenStart = tokenEnd = 0;
+   }
+
+   if (tokenEnd == 0) {
+      tokenEnd = lastChar + 1;
+   }
+
+   Process_updateCmdline(process, command, tokenStart, tokenEnd);
+
+   // comm is set in CygwinProcessTable_recurseProcTree from /proc/[pid]/stat
+
+   char filename[MAX_NAME + 1];
+
+   /* execve could change /proc/[pid]/exe, so procExe should be updated */
+#if defined(HAVE_READLINKAT) && defined(HAVE_OPENAT)
+   amtRead = readlinkat(procFd, "exe", filename, sizeof(filename) - 1);
+#else
+   amtRead = Compat_readlink(procFd, "exe", filename, sizeof(filename) - 1);
+#endif
+   if (amtRead > 0) {
+      filename[amtRead] = 0;
+      if (!process->procExe ||
+          (!process->procExeDeleted && !String_eq(filename, process->procExe)) ||
+          process->procExeDeleted) {
+
+         bool oldExeDeleted = process->procExeDeleted;
+
+         process->procExeDeleted = access(filename, F_OK) < 0;
+
+         if (oldExeDeleted != process->procExeDeleted)
+            process->mergedCommand.lastUpdate = 0;
+
+         Process_updateExe(process, filename);
+      }
+   } else if (process->procExe) {
+      Process_updateExe(process, NULL);
+      process->procExeDeleted = false;
+   }
+
+   return true;
+}
```

Comment:
Have a look [here](https://stackoverflow.com/questions/6530565/getting-another-process-command-line-in-windows).
Also note the [comments by Reymond Chen on the subject](https://devblogs.microsoft.com/oldnewthing/20091125-00/?p=15923).

When rewriting the code, make sure the `mergedCommand` fields are properly set.
The fields `procExeDeleted` and `usesDeletedLib` can be assumed `false` on Windows, due to [$reasons](https://learn.microsoft.com/en-us/previous-versions/technet-magazine/cc983682(v=msdn.10)?redirectedfrom=MSDN) …

Reply:
i am not familiar with windows apis... so i dont have plan to replace these with native api calls when cygwin already provides wrappers for it.